sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -567,6 +567,26 @@ object WholeStageCodegenExec {
   def isTooManyFields(conf: SQLConf, dataType: DataType): Boolean = {
     numOfNestedFields(dataType) > conf.wholeStageMaxNumFields
   }
+
+  // The whole codegen generates java code on the driver side and sends it to the Executor side
+  // for execution after compilation. The whole codegen can bring significant performance
+  // improvements in large data and distributed environments. However, in the test environment,
+  // due to the small amount of data, the time to generate Java code takes up a major part of the
+  // entire runtime. So we summarize the total generation time and output it to the execution log
+  // for easy analysis and view.
+  private val _generateJavaTime = new LongAccumulator
```

Comment:
why do we need accumulator if it's a pure driver side thing?

Reply:
I just followed #28081

Reply:
#28081 is different as we compile the code at the executor side. Yea let's use `AtomicLong`

Reply:
@cloud-fan I learned the use of `LongAccumulator` for statistics in a distributed environment.Thanks!